packages/providers/src/call-retrier.ts

```diff
@@ -0,0 +1,65 @@
+import { sleep } from './utils';
+
+type Backoff = 'linear' | 'exponential' | 'fixed';
+
+export interface RetryOptions {
+  /**
+   * Amount of attempts to retry before failing the call.
+   */
+  maxAttempts: number;
+  /**
+   * Backoff strategy to use when retrying.
+   */
+  backoff: Backoff;
+  /**
+   * Base duration for backoff strategy.
+   */
+  baseDuration: number;
+}
+
+function getWaitDuration(options: RetryOptions, attempt: number) {
+  if (attempt === 0) {
+    return options.baseDuration;
+  }
+
+  switch (options.backoff) {
+    case 'linear':
+      return options.baseDuration * attempt;
+    case 'exponential':
+      return options.baseDuration * (2 ^ (attempt - 1));
+    case 'fixed':
+      return options.baseDuration;
+    default:
+      throw new Error();
```

Comment:
A default strategy might be nice here to reduce the implementation on the consumer.

Reply:
That's cool but too complicated—steps for simplifying:
 1. Internalize the `baseDuration`
 1. Use `exponential` as the default
 1. Remove everything else

Before:

```ts
const retryOptions = { maxAttempts: 5, baseDuration: 150, backoff: 'fixed' };
const provider = await Provider.create(FUEL_NETWORK_URL, { retryOptions });
```

After:

```ts
const provider = await Provider.create(FUEL_NETWORK_URL, { maxRetries: 5 });
```

Reply:
> 1. Internalize the baseDuration

I didn't quite get this part. Did you mean to not allow the user specify a `baseDuration`?

> 2. Use exponential as the default

But still allow the other two?

> 3. Remove everything else

I don't see anything else besides `baseDuration` and `backoff` and `maxAttempts` (to be renamed into `maxRetries`). What did you mean by this?

Reply:
I did the following:
- Renamed `maxAttempts` into `maxRetries`,
- Made `baseDuration` optional with a default of 150ms,
- Made `backoff` optional with the default being `exponential`

Reply:
I believe you got into the flow and overengineered it a bit.

1. Yes.
2. No, only `maxRetries`
3. Everything else = `baseDuration`, `backoff`

Reply:
As we just spoke, although I think it's a bit over, and that's ok.

We can revisit things as we go if necessary.